src/AppBundle/Command/AddUserCommand.php

```diff
@@ -0,0 +1,259 @@
+            $question->setValidator(function ($answer) {
+                if (empty($answer)) {
+                    throw new \RuntimeException('The username cannot be empty');
+                }
+
+                return $answer;
+            });
+            $question->setMaxAttempts(self::MAX_ATTEMPTS);
+
+            $username = $console->ask($input, $output, $question);
+            $input->setArgument('username', $username);
+        } else {
+            $output->writeln(' > <info>Username</>: '.$username);
+        }
+
+        // Ask for the password if it's not defined
+        if (null === $password = $input->getArgument('password')) {
+            $question = new Question(' > <info>Password</> (your type will be hidden): ');
+            $question->setValidator(array($this, 'passwordValidator'));
+            $question->setHidden(true);
+            $question->setMaxAttempts(self::MAX_ATTEMPTS);
+
+            $password = $console->ask($input, $output, $question);
+            $input->setArgument('password', $password);
+        } else {
+            $output->writeln(' > <info>Password</>: '.str_repeat('*', strlen($password)));
+        }
+
+        // Ask for the email if it's not defined
+        if (null === $email = $input->getArgument('email')) {
+            $question = new Question(' > <info>Email</>: ');
+            $question->setValidator(array($this, 'emailValidator'));
+            $question->setMaxAttempts(self::MAX_ATTEMPTS);
+
+            $email = $console->ask($input, $output, $question);
+            $input->setArgument('email', $email);
+        } else {
+            $output->writeln(' > <info>Email</>: '.$email);
+        }
+    }
+
+    /**
+     * This method is executed after the interact() and before the execute()
+     * method. It's main purpose is to initialize the variables used in the rest
+     * of the command methods.
+     */
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        $this->em = $this->getContainer()->get('doctrine')->getManager();
+    }
+
+    /**
+     * This method is executed after interact() and initialize(). It usually
+     * contains the logic to execute to complete this command task.
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $startTime = microtime(true);
+
+        // first check if a user with the same username already exists
+        $existingUser = $this->em->getRepository('AppBundle:User')->findOneBy(array(
+            'username' => $username = $input->getArgument('username'),
+        ));
+
+        if (null !== $existingUser) {
+            throw new \RuntimeException(sprintf('There is already a user registered with the "%s" username.', $username));
+        }
+
+        $this->username = $input->getArgument('username');
+        $this->plainPassword = $input->getArgument('password');
+        $this->email = $input->getArgument('email');
```

Comment:
I suggest using local variables for these instead of storing them in the command instance